crates/daphne-server/tests/e2e/e2e.rs

```diff
@@ -157,6 +158,11 @@ async fn leader_upload(version: DapVersion) {
     let client = t.http_client();
     let hpke_config_list = t.get_hpke_configs(version, client).await.unwrap();
     let path = t.upload_path();
+    let method = if version == DapVersion::Draft09 {
+        &http::Method::PUT
+    } else {
+        &http::Method::POST
+    };
```

Comment:
Defensive programming: Use `match` so that if add variants in the future, the compiler tells you if you failed to handle a variant
```suggestion
    let method = match version {
        DapVersion::Draft09 =>  &http::Method::PUT,
        DavVersion::DraftLatest => &http::Method::POST,
    };
```